template/table.py

```diff
@@ -8,31 +8,50 @@
 TIMESTAMP_COLUMN = 2
 SCHEMA_ENCODING_COLUMN = 3
 
+
 class Base_page:
-        # Array of Pages num_columns long plus schema and indirections columns
-        # size() - 2 = schema column;  size() - 1 indirection column
-        def __init__(self, num_columns):
-            self.columns = [Page() for i in range(num_columns + 2)] 
+    '''
+    :param tail_page_list: list      Stores a list of all Tail_Page objects associated with the Base_page
+    :param columns_list: list        Stores a list of all Page objects for the Base_page, these are the data columns
+    :param tail_page_count: int      An integer that holds the current amount of tail pages the Base_page has
+    '''
+    def __init__(self, num_columns: int):
+        # Create a starting Tail Page for updates
+        self.tail_page_list = [Tail_page(num_columns)]
+        # Create a list of Physical Pages num_columns long plus Indirection, RID, TimeStamp, and Schema columns
+        self.columns_list = [Page() for i in range(num_columns + 4)]
+        self.tail_page_count = 1
+
+
+class Tail_page:
+    '''
+    :param columns_list: list       Stores a list of all Page objects for the Tail_page, these are the data columns
+    '''
+    def __init__(self, num_columns: int):
+        # Create a list of Physical Pages num_columns long plus Indirection, RID, TimeStamp, and Schema columns
+        self.columns_list = [Page() for i in range(num_columns + 4)]
+        
     
 class Page_range:
-    # Array of Base_pages based on Const.BASE_PAGE_COUNT
-    def __init__(self, num_columns):
+    '''
+    :param pages: list      Stores a list of all Base_page objects in the Page_range
+    '''
+    def __init__(self, num_columns: int):
+        # Array of Base_pages based on Const.BASE_PAGE_COUNT
         self.pages = [Base_page(num_columns) for i in range(BASE_PAGE_COUNT)]
 
-class Record:
```

Comment:
I think for query we will need a record schema so on Sunday we can discuss what the record class needs

Reply:
I can work on schema implementation later since it should only matter at the table scope